whois-api/src/main/java/net/ripe/db/whois/api/rest/WhoisServletDeployer.java

```diff
@@ -104,6 +106,9 @@ public void deploy(WebAppContext context) {
         jaxbJsonProvider.configure(SerializationFeature.WRAP_ROOT_VALUE, false);
         resourceConfig.register(jaxbJsonProvider);
 
+        final MessageBodyWriter<WhoisResources> customMessageBodyWriter = new WhoisResourcesPlainTextWriter();
+        resourceConfig.register(customMessageBodyWriter);
+
```

Comment:
This servlet deployer is being used for lot of other services and not every service returns whoisResources. Also this issue exists only when we throw exception else streamingMarshal takes care of it. Check 404 responses are handled properly. May be the best place to make changes is in RestServiceHelper where we create errorEntity for whoisResources but it's just a suggestion.

Reply:
yes, you are right, other services that uses whoisResources are affected (some IT breaks because of that). To fix that I did next modification;

   @Override
     public boolean isWriteable(Class<?> aClass, Type type, Annotation[] annotations, MediaType mediaType) {
         return aClass == WhoisResources.class && MediaType.TEXT_PLAIN.contains(mediaType.getType());
     }
     
Now we are checking the class and the MediaType...not just the class

Reply:
I will create a new PR with this change. However, it will be a very big change

Reply:
https://github.com/RIPE-NCC/whois/pull/1136 -> Another approach
1) We are not able to throw the errors as we do for the 404, because we throw those errors from the Mashal strategy (StreamingMarshalTextPlainText). If we want to do this we need to move all the logic that we have in whoisRestService.lookup and whoisSeachService to that Marshal strategy and perform a big try catch to return the error in the correct format.
2) Jackson is able to transform the error in xml or json using @XmlElement and @JsonProperty but it does not support plain/text. This is why we don't have this problem with XML.

Reply:
I prefer this approach @MiguelAHM

Reply:
ok, applied your requested changes